File: src/main/Python/boj_step9/boj9506.py
import sys
cin = sys.stdin.readline

while True:
    n = int(cin())
    factors = []
    if n == -1: exit()
    for i in range(1, n // 2 + 1):
        if n % i == 0:
            factors.append(i)
    if n == sum(factors):
        print(n, '=', ' + '.join(map(str, factors)))
    else:
        print(f'{n} is NOT perfect.')

File: src/main/Cpp/boj_step9/boj9506.cpp
#include <iostream>
#include <list>
#include <numeric>
using namespace std;

int main() {
    list<int> factors;
    int n;
    cin >> n;

    while (n != -1) {
        for (int devisor = 1; devisor <= n / 2; devisor++)
            if (n % devisor == 0)
                factors.push_back(devisor);
        if (n == accumulate(factors.begin(), factors.end(), 0)) {
            cout << n << " = ";
            for (size_t i = 0; i < factors.size(); ++i) {
                if (i > 0) cout << " + ";
                cout << factors.front();
                factors.pop_front();
            }
            cout << endl;
        } else
            cout << n << " is NOT perfect." << endl;
    }
}